crates/requirements-txt/src/lib.rs

```diff
@@ -947,6 +972,46 @@ impl From<io::Error> for RequirementsTxtParserError {
     }
 }
 
+/// Calculates the column and line offset of a given cursor based on the
+/// number of Unicode codepoints.
+fn to_row_column(content: &str, position: usize) -> (usize, usize) {
+    let mut line = 1;
+    let mut column = 1;
+
+    let mut chars = content.char_indices().peekable();
+    while let Some((index, char)) = chars.next() {
+        if index >= position {
+            break;
+        }
+        match char {
+            '\r' => {
+                // If the next character is a newline, skip it.
```

Comment:
Tweaked this a little because `\r` on its own _should_ be considered a newline (it's not used on (m)any modern platforms, but older macOS did use it).

Reply:
Np, I went a bit back and forth on that. Glad it's settled

Reply:
Sorry for all the back-and-forth, I just have scars from Ruff where we didn't handle all the newline kinds and eventually hit panics in certain projects.

Reply:
no worries, makes sense, and feedback is always welcomed 💯